rotkehlchen/chain/ethereum/node_inquirer.py

```diff
@@ -223,6 +237,36 @@ def _ens_lookup(
             log.error(f'Error deserializing address {address}')
             return None
 
+    @protect_with_lock()
+    def assure_curve_protocol_cache_is_queried(self) -> bool:
+        """
+        Make sure that curve information that needs to be queried is queried and if not query it.
+        Returns true if the cache was modified or false otherwise.
+
+        1. Deletes all previous cache values
+        2. Queries information about curve pools' addresses, lp tokens and used coins
+        3. Saves queried information in the cache in globaldb
+        """
+        if should_update_protocol_cache(GeneralCacheType.CURVE_LP_TOKENS) is False:
+            return False
+
+        curve_address_provider = self.contracts.contract('CURVE_ADDRESS_PROVIDER')
+        # Using savepoint to not end up having partially populated cache
+        with GlobalDBHandler().conn.savepoint_ctx() as savepoint_cursor:
```

Comment:
I know it is not related to this PR's changes, but:

Currently this savepoint will stay open for a long time (because there are remote queries).

Both `update_curve_registry_pools_cache` and `update_curve_metapools_cache` write to the db only at the very end.

So maybe change this to:
1. Query everything
2. Quickly clear the cache and write new pools

Reply:
Thanks for the suggestion. Let me check.

Reply:
here: https://github.com/rotki/rotki/pull/5498